receiver/datadogreceiver/internal/translator/sketches.go

```diff
@@ -0,0 +1,266 @@
+// Note that negative indices in exponential histograms do not necessarily correspond to negative values; they correspond with values between 0 and 1,
+// on either the negative or positive side
+func mapSketchBucketsToHistogramBuckets(sketchKeys []int32, sketchCounts []uint32) (map[int]uint64, map[int]uint64, uint64) {
+	var zeroCount uint64
+
+	var positiveBuckets = make(map[int]uint64)
+	var negativeBuckets = make(map[int]uint64)
+
+	// The data format for the sketch received from the sketch payload does not have separate positive and negative buckets,
+	// and instead just uses a single list of sketch keys that are in order by increasing bucket index, starting with negative indices,
+	// which correspond to negative buckets
+	for i := range sketchKeys {
+		if sketchKeys[i] == 0 { // A sketch key of 0 corresponds to the zero bucket
+			zeroCount += uint64(sketchCounts[i])
+			continue
+		}
+
+		// The approach here is to use the Datadog sketch index's lower bucket boundary to find the
+		// OTel exponential histogram bucket that with the closest range to the sketch bucket. Then,
+		// the buckets before and after that bucket are also checked for overlap with the sketch bucket.
+		// A count proportional to the intersection of the sketch bucket with the OTel bucket(s) is then
+		// added to the OTel bucket(s). After looping through all possible buckets that are within the Sketch
+		// bucket range, the bucket with the highest proportion of overlap is given the remaining count
+		sketchLowerBound, sketchUpperBound := getSketchBounds(sketchKeys[i])
+		sketchBucketSize := sketchUpperBound - sketchLowerBound
+		histogramKey := sketchLowerBoundToHistogramIndex(sketchLowerBound)
+		highestCountProportion := 0.0
+		highestCountIdx := 0
+		targetBucketCount := uint64(sketchCounts[i])
+		var currentAssignedCount uint64
+
+		//TODO: look into better algorithms for applying fractional counts
+		for outIndex := histogramKey; histogramLowerBound(outIndex) < sketchUpperBound; outIndex++ {
+			histogramLowerBound, histogramUpperBound := getHistogramBounds(outIndex)
+			lowerIntersection := math.Max(histogramLowerBound, sketchLowerBound)
+			higherIntersection := math.Min(histogramUpperBound, sketchUpperBound)
+
+			intersectionSize := higherIntersection - lowerIntersection
+			proportion := intersectionSize / sketchBucketSize
+			if proportion <= 0 {
+				continue // In this case, the bucket does not overlap with the sketch bucket, so continue to the next bucket
+			}
+			if proportion > highestCountProportion {
+				highestCountProportion = proportion
+				highestCountIdx = outIndex
+			}
+			// OTel exponential histograms only support integer bucket counts, so rounding needs to be done here
+			roundedCount := uint64(proportion * float64(sketchCounts[i]))
+			if sketchKeys[i] < 0 {
+				negativeBuckets[outIndex] += roundedCount
```

Comment:
Are we sure negative values can be observed in sketches?  The zero threshold only makes sense if that's not allowed.

Reply:
We've looked at the DD agent code and it seems like sketches coming into the agent have separate negative and positive buckets (list of K, N values) with a very high precision https://github.com/DataDog/sketches-go/blob/7546f8f95179bb41d334d35faa281bfe97812a86/ddsketch/mapping/logarithmic_mapping.go#L48 . 

However the agent makes a single list of K, N pairs out of these two lists by adding any K<=0 to the zero bucket and putting positive buckets as positive Ks into the new list, while putting negative buckets as negative Ks into the list.

That means that the original negative K values corresponding to high precision buckets (e^(K-1338)) where K<=0 are not represented, thus the size and threshold of the zero bucket is indeed e^((1-1338)/(1/gamma)), which is about 9.941..E-10.

Conversely , testing K=1 gives this lower exponential bucket boundary:
```
>>> print(math.pow(math.pow(2.0, math.pow(2.0, -5)), -957))
9.938519454345803e-10
```